docs/tpm-simulators.md

```diff
@@ -0,0 +1,162 @@
+# TPM simulators
+
+It is possible to use TPM simulators instead of real TPMs **for testing purposes
+only**. Two simulators were tested and are known to be working with Fobnail:
+[Microsoft's](https://github.com/microsoft/ms-tpm-20-ref) and
+[IBM's](https://sourceforge.net/projects/ibmswtpm2/). Refer to their respective
+documentations for build and launch instructions.
+
+## Differences between simulated and real TPM
+
+The main difference is that simulator is started after system is up. This means
+that there will be no measurements of firmware nor OS. Attestation is based on
+these measurements, without them there is nothing to attest to, which makes use
+of simulated TPM useless for practical applications. It also means that nothing
+called `TPM2_Startup` command. Attester is able to detect this and calls that
+function automatically.
+
+Hardware and firmware TPM vendors give assurance of proper functioning of their
+TPMs. This is done by creating a certificate for each TPM separately,
+specifically for their Endorsement Keys (EKs). These certificates are created
+during TPM manufacturing and provisioning process, and are written to NVRAM.
+They point to their signing certificates through `authorityInfoAccess` X509V3
+extension. The chain continues until a self-signed root that is implicitly
+trusted. That root must be known by Fobnail Token in advance in order to trust
+given TPM.
+
+For simulated TPM, such certificate has to be manually created and injected into
+NVRAM. EK certificate is created for EK created from EPS (Endorsement Primary
+Seed), which is randomly created when the simulator is started for the first
+time. Nobody is issuing certificates for all instances of simulated TPMs. One of
+the reasons is probably the fact that emulators are not fully compatible with
+specification, especially when it comes to hardware protection mechanisms
+(Physical Presence, protected storage etc.).
+
+There are also other small differences, e.g. after non-orderly shutdown `safe`
+field in `TPMS_CLOCK_INFO` structure is set after 12 seconds in simulator, while
+specification allows for up to 2^22 milliseconds (around 70 minutes) to help
+with NVRAM wear leveling. Note that this field is checked by Fobnail during
+attestation.
+
+## Provisioning simulated TPM
+
+As mentioned, EK certificate must be written to TPM NVRAM. In addition, it must
+point to valid CA certificate that will be downloaded during platform
+provisioning. A script for simplifying this process and required configuration
+are included in [Attester's repository](https://github.com/fobnail/fobnail-attester/tree/main/tools).
+
+Configuration assumes that CA certificate will be made available at address
+`http://127.0.0.1:8080/ca_cert.der`, to change this please edit `ek_v3.ext` in
+`tools` directory.
```

Comment:
:thinking: Did I already create ca_cert.der while going through previous steps in the building section? Or TL;DR in keys_and_certifcates?

Those instructions are created for Linux users, so we can safely assume that the user would like to simply replicate ca_cert.der exposure, so why not tell him/her about `python3 -m http.server 8080` (I guess this is what the author used)?

Reply:
So this paragraph just bothers a user who is confused about if he/she should take action or if everything is ok, and the hello world provisioning example would work with what he/she has so far. We can safely move that to the latter text after success would be accomplished, and the user then may consider more complex stuff like a change in configuration.

Reply:
Moved down in https://github.com/fobnail/docs/pull/47/commits/24c61127cf5e79c93abf14ba615b5a7e7567cb5b